cmd/thanos/query-frontend.go

```diff
@@ -111,51 +92,49 @@ func runQueryFrontend(
 	logger log.Logger,
 	reg *prometheus.Registry,
 	tracer opentracing.Tracer,
-	conf *queryFrontendConfig,
+	cfg *config,
 	comp component.Component,
 ) error {
+	cacheConfContentYaml, err := cfg.cachePathOrContent.Content()
+	if err != nil {
+		return err
+	}
+	if len(cacheConfContentYaml) > 0 {
+		if err := yaml.UnmarshalStrict(cacheConfContentYaml, &cfg.QueryRange.ResultsCacheConfig.CacheConfig); err != nil {
+			return errors.Wrap(err, "parsing cache config YAML file")
+		}
+	}
 
-	if len(conf.downstreamURL) == 0 {
-		return errors.New("downstream URL should be configured")
+	err = cfg.Validate()
+	if err != nil {
+		return errors.Wrap(err, "error validating the config")
 	}
 
 	fe, err := frontend.New(frontend.Config{
-		DownstreamURL:        conf.downstreamURL,
-		CompressResponses:    conf.compressResponses,
-		LogQueriesLongerThan: conf.LogQueriesLongerThan,
+		DownstreamURL:        cfg.Frontend.DownstreamURL,
+		CompressResponses:    cfg.Frontend.CompressResponses,
+		LogQueriesLongerThan: cfg.Frontend.LogQueriesLongerThan,
 	}, logger, reg)
 	if err != nil {
 		return errors.Wrap(err, "setup query frontend")
 	}
 	defer fe.Close()
 
-	limits := queryfrontend.NewLimits(
-		conf.queryRangeConfig.maxQueryParallelism,
-		time.Duration(conf.queryRangeConfig.maxQueryLength),
-		conf.queryRangeConfig.cacheMaxFreshness,
-	)
-
-	respCacheContentYaml, err := conf.queryRangeConfig.respCacheConfig.Content()
+	limits, err := validation.NewOverrides(validation.Limits{
+		MaxQueryLength:      cfg.Limits.MaxQueryLength,
+		MaxQueryParallelism: cfg.Limits.MaxQueryParallelism,
+		MaxCacheFreshness:   cfg.Limits.MaxCacheFreshness,
+	}, nil)
 	if err != nil {
```

Comment:
Is it necessary to check error here. `validation.NewOverrides` always returns nil.

Reply:
We don't know that this will remain like this. The implementation can change.

Reply:
Ahh, makes sense.